Sources/TuistLoader/Models+ManifestMappers/TestAction+ManifestMapper.swift

```diff
@@ -9,22 +9,50 @@ extension TuistCore.TestAction {
     ///   - manifest: Manifest representation of test action model.
     ///   - generatorPaths: Generator paths.
     static func from(manifest: ProjectDescription.TestAction, generatorPaths: GeneratorPaths) throws -> TuistCore.TestAction {
-        let targets = try manifest.targets.map { try TuistCore.TestableTarget.from(manifest: $0, generatorPaths: generatorPaths) }
-        let arguments = manifest.arguments.map { TuistCore.Arguments.from(manifest: $0) }
-        let configurationName = manifest.configurationName
-        let coverage = manifest.coverage
-        let codeCoverageTargets = try manifest.codeCoverageTargets.map {
-            TuistCore.TargetReference(projectPath: try generatorPaths.resolveSchemeActionProjectPath($0.projectPath),
-                                      name: $0.targetName)
+        let testPlans: [TuistCore.TestPlan]?
+        let targets: [TuistCore.TestableTarget]
+        let arguments: TuistCore.Arguments?
+        let coverage: Bool
+        let codeCoverageTargets: [TuistCore.TargetReference]
+        let diagnosticsOptions: Set<TuistCore.SchemeDiagnosticsOption>
+        let language: String?
+        let region: String?
+
+        if let plans = manifest.testPlans {
+            let defaultPlan = try TuistCore.TestPlan.from(path: plans.default, isDefault: true, generatorPaths: generatorPaths)
+            let otherPlans = try plans.other.map { try TuistCore.TestPlan.from(path: $0, isDefault: false, generatorPaths: generatorPaths) }
+
+            testPlans = [defaultPlan] + otherPlans
+
+            targets = []
+            arguments = nil
+            coverage = false
+            codeCoverageTargets = []
+
+            diagnosticsOptions = Set()
+            language = nil
+            region = nil
+        } else {
+            targets = try manifest.targets.map { try TuistCore.TestableTarget.from(manifest: $0, generatorPaths: generatorPaths) }
+            arguments = manifest.arguments.map { TuistCore.Arguments.from(manifest: $0) }
+            coverage = manifest.coverage
+            codeCoverageTargets = try manifest.codeCoverageTargets.map {
+                TuistCore.TargetReference(projectPath: try generatorPaths.resolveSchemeActionProjectPath($0.projectPath),
+                                          name: $0.targetName)
+            }
+
+            diagnosticsOptions = Set(manifest.diagnosticsOptions.map { TuistCore.SchemeDiagnosticsOption.from(manifest: $0) })
+            language = manifest.language
+            region = manifest.region
+
+            testPlans = []
```

Comment:
I believe this needs to be set to `nil` (the reason why the acceptance tests were failing)

When set to an empty array, Xcode scheme switches to test plan mode with empty list as opposed to the previous behaviour with the explicit test targets being specified.

![Screenshot 2020-10-26 at 07 36 29](https://user-images.githubusercontent.com/11914919/97145350-fbde9d80-175d-11eb-88c8-1e6c0a39b670.png)

You can locally test this by generating `fixtures/ios_app_with_custom_scheme` inspect the `Workspace-App` scheme.

Reply:
thanks. changed it.